source/gist/multiset-addendum-question.lagda

```diff
@@ -0,0 +1,203 @@
+Alice Laroche, 14th June 2024
+
+This file answers the question asked in Iterative.Multisets-Addendum
+That is : Is there a function Πᴹ of the above type that satisfies the
+following equation?
+
+Σ Πᴹ ꞉ ((X → 𝕄) → 𝕄)
+     , ((A : X → 𝕄) → Πᴹ A ＝ ssup
+                               (Π x ꞉ X , 𝕄-root (A x))
+                               (λ g → Πᴹ (λ x → 𝕄-forest (A x) (g x))))
+
+We prove that it isn't the case in general, as the existence of function for
+the empty type would allow infinite recursion.
+We then prove that the function exists up to equivalence for pointed types,
+and, admitting function extensionality, for inhabited types.
+
+
+\begin{code}
+
+{-# OPTIONS --safe --without-K --exact-split #-}
+
+open import MLTT.Spartan
+open import UF.Base
+open import UF.Equiv
+open import UF.FunExt
+open import UF.PropTrunc
+open import UF.Subsingletons
+open import UF.Univalence
+open import W.Type
+
+module gist.multiset-addendum-question
+        (ua : Univalence)
+        (𝓤 : Universe)
+       where
+
+open import Iterative.Multisets 𝓤
+open import Iterative.Multisets-Addendum ua 𝓤
+
+swap-Idtofun : {X Y : 𝓤 ̇ } {Z : 𝓥 ̇ } → {f : X → Z} {g : Y → Z}
+             → (p : Y ＝ X)
+             → f ∘ Idtofun p ＝ g
+             → f ＝ g ∘ Idtofun⁻¹ p
+swap-Idtofun  refl refl = refl
+
+Question𝟘 :
+ ¬ (Σ Πᴹ ꞉ ((𝟘 {𝓤} → 𝕄) → 𝕄)
+         , ((A : 𝟘 → 𝕄) → Πᴹ A ＝ ssup
+                                  (Π x ꞉ 𝟘 , 𝕄-root (A x))
+                                  (λ g → Πᴹ (λ x → 𝕄-forest (A x) (g x)))))
+Question𝟘 (Πᴹ , eq) = recurs A (Πᴹ A) (eq A)
+ where
+  A : 𝟘 → 𝕄
+  A x = 𝟘-elim x
+
+  recurs : (A : 𝟘 → 𝕄) → (x : 𝕄)
```

Comment:
@martinescardo This reminds me that it might be a good idea to add a bit more details to the clause on this in the contribution guidelines.

When the type declaration `f : X → Y → Z` is broken into separate lines, the convention is to indent it like:

```agda
f : X
  → Y
  → Z
```

as to avoid arrows on the same line.

What do we do if we have something like the function below?

```agda
f : (x x' : X) (y : Y(x)) → Z(x, x', y)
```

Is it better to write this as

```agda
f : (x x' : X)
  → (y : Y(x))
  → (z : Z(x, x', y))
```

or as

```agda
f : (x x' : X) (y : Y(x))
  → Z(x, x', y)
```

as to avoid multiple quantifications on the same line?

I sometimes have a hard time deciding between the two. If either of these is a more standard convention, I would be happy to write it down for future reference.